src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -49,6 +62,107 @@ public AWSResourceQueryer(IAWSClientFactory awsClientFactory)
+            }
+
+            return environment.Environments.First();
+        }
+
+        public async Task<Amazon.ElasticLoadBalancingV2.Model.LoadBalancer> DescribeElasticLoadBalancer(string loadBalancerArn)
+        {
+            var elasticLoadBalancingClient = _awsClientFactory.GetAWSClient<IAmazonElasticLoadBalancingV2>();
+
+            var loadBalancers = await elasticLoadBalancingClient.DescribeLoadBalancersAsync(new DescribeLoadBalancersRequest
+            {
+                LoadBalancerArns = new List<string> { loadBalancerArn }
+            });
+
+            if (!loadBalancers.LoadBalancers.Any())
+            {
+                throw new AWSResourceNotFoundException($"The load balancer '{loadBalancerArn}' does not exist.");
+            }
+
+            return loadBalancers.LoadBalancers.First();
+        }
+
+        public async Task<List<Amazon.ElasticLoadBalancingV2.Model.Listener>> DescribeElasticLoadBalancerListeners(string loadBalancerArn)
+        {
+            var elasticLoadBalancingClient = _awsClientFactory.GetAWSClient<IAmazonElasticLoadBalancingV2>();
+
+            var listeners = await elasticLoadBalancingClient.DescribeListenersAsync(new DescribeListenersRequest
+            {
+                LoadBalancerArn = loadBalancerArn
+            });
+
+            if (!listeners.Listeners.Any())
+            {
+                throw new AWSResourceNotFoundException($"The load balancer '{loadBalancerArn}' does not have any listeners.");
+            }
+
+            return listeners.Listeners;
+        }
+
+        public async Task<DescribeRuleResponse> DescribeCloudWatchRule(string ruleName)
+        {
+            var cloudWatchEventsClient = _awsClientFactory.GetAWSClient<IAmazonCloudWatchEvents>();
+
+            var rule = await cloudWatchEventsClient.DescribeRuleAsync(new DescribeRuleRequest
+            {
+                Name = ruleName
+            });
+
+            if (rule == null)
+            {
+                throw new AWSResourceNotFoundException($"The CloudWatch rule'{ruleName}' does not exist.");
+            }
+
+            return rule;
+        }
+
+        public async Task<string> GetS3BucketLocation(string bucketName)
+        {
+            if (string.IsNullOrEmpty(bucketName))
+            {
+                throw new ArgumentNullException($"The bucket name is null or empty.");
+            }
+
+            var s3Client = _awsClientFactory.GetAWSClient<IAmazonS3>();
+
+            var location = await s3Client.GetBucketLocationAsync(bucketName);
+
+            var region = "";
+            if (location.Location.Equals(S3Region.USEast1))
+                region = "us-east-1";
+            else if (location.Location.Equals(S3Region.EUWest1))
+                region = "eu-west-1";
+            else
+                region = location.Location;
+
+            return region;
```

Comment:
```suggestion
            switch (location.Location) 
            {
                case S3Region.USEast1:
                    return RegionEndpoint.USEast1.SystemName;
                case S3Region.EUWest1:
                    return RegionEndpoint.EUWest1.SystemName;
                default:
                    return location.Location;
            }
```

Reply:
I couldn't use S3Region in a switch because it is not a constant value.

Reply:
This is in S3 nuget 

https://github.com/aws/aws-sdk-net/blob/09bd2aeebf743d70c29216a9f731a1a580f3b516/sdk/src/Services/S3/Generated/S3Enumerations.cs#L23

Reply:
Yeah I still wasn't able because it is not a const